sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-data.js

```diff
@@ -181,19 +220,39 @@ function generateTest(internalFormat, pixelFormat, pixelType, prologue, resource
     }
 
     function runTestOnBindingTarget(bindingTarget, program) {
+        var k = blackColor;
+        var r = redColor;
+        var g = greenColor;
         var cases = [
-            { sub: false, flipY: true, premultiplyAlpha: false },
-            { sub: false, flipY: false, premultiplyAlpha: false },
-            { sub: false, flipY: true, premultiplyAlpha: true },
-            { sub: false, flipY: false, premultiplyAlpha: true },
-            { sub: true, flipY: true, premultiplyAlpha: false },
-            { sub: true, flipY: false, premultiplyAlpha: false },
-            { sub: true, flipY: true, premultiplyAlpha: true },
-            { sub: true, flipY: false, premultiplyAlpha: true },
+            { expected: [r, r, g, g], flipY: false, premultiplyAlpha: false, sub: false },
+            { expected: [r, r, g, g], flipY: false, premultiplyAlpha: false, sub: true },
+            { expected: [r, k, g, k], flipY: false, premultiplyAlpha: true, sub: false },
+            { expected: [r, k, g, k], flipY: false, premultiplyAlpha: true, sub: true },
+            { expected: [g, g, r, r], flipY: true, premultiplyAlpha: false, sub: false },
+            { expected: [g, g, r, r], flipY: true, premultiplyAlpha: false, sub: true },
+            { expected: [g, k, r, k], flipY: true, premultiplyAlpha: true, sub: false },
+            { expected: [g, k, r, k], flipY: true, premultiplyAlpha: true, sub: true },
         ];
 
+        if (wtu.getDefault3DContextVersion() > 1) {
+            var morecases = [];
+            // make 4 copies of the original case, one for each corner
```

Comment:
Comment needs to be updated: "make 2 copies of the original case for the opposite corners" or similar.